core/src/test/java/org/elasticsearch/cluster/metadata/MetaDataTests.java

```diff
@@ -41,4 +41,61 @@ public void testIndexAndAliasWithSameName() {
         }
     }
 
+    public void testResolveIndexRouting() {
+        IndexMetaData.Builder builder = IndexMetaData.builder("index")
+                .settings(Settings.builder().put(IndexMetaData.SETTING_VERSION_CREATED, Version.CURRENT))
+                .numberOfShards(1)
+                .numberOfReplicas(0)
+                .putAlias(AliasMetaData.builder("alias0").build())
+                .putAlias(AliasMetaData.builder("alias1").routing("1").build())
+                .putAlias(AliasMetaData.builder("alias2").routing("1,2").build());
+        MetaData metaData = MetaData.builder().put(builder).build();
+        assertEquals(metaData.resolveIndexRouting(null, null, null), null);
+        assertEquals(metaData.resolveIndexRouting(null, "0", null), "0");
+        assertEquals(metaData.resolveIndexRouting(null, "0", "index"), "0");
+        assertEquals(metaData.resolveIndexRouting(null, null, "alias0"), null);
+        assertEquals(metaData.resolveIndexRouting(null, "0", "alias0"), "0");
+        assertEquals(metaData.resolveIndexRouting("32", null, "alias0"), "32");
+        assertEquals(metaData.resolveIndexRouting("32", "0", "alias0"), "0");
+
+        assertEquals(metaData.resolveIndexRouting(null, null, "alias1"), "1");
+        assertEquals(metaData.resolveIndexRouting("32", null, "alias1"), "1");
+        assertEquals(metaData.resolveIndexRouting("32", "1", "alias1"), "1");
+        try {
+            metaData.resolveIndexRouting(null, "0", "alias1");
+            fail("should fail");
+        } catch (IllegalArgumentException ex) {
+            // Expected
+        }
+        try {
+            metaData.resolveIndexRouting("32", "0", "alias1");
+            fail("should fail");
+        } catch (IllegalArgumentException ex) {
+            // Expected
```

Comment:
we should check the exception is what we expect (assert part or all of the message)